crates/iceberg-catalog/src/catalog/views.rs

```diff
@@ -30,7 +31,7 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
         validate_namespace_ident(&namespace)?;
 
         // ------------------- AUTHZ -------------------
-        A::check_list_tables(
```

Comment:
I am thinking about just having a "check_list_tabular" and not separate between tables and views.
Initially I thought this would suffice, as Databases also only have a grant "CREATE", ... on a Namespace, that allows to create / read / ... any type of Object (View, Materialized View, Tabular,...) that supports this capability.

However, I came to the conclusion that implementing it with only one "tabular" type might be more complicated, as possible grants will differ in the future between views and tables (i.e. insert). Thus, the AuthZ system will have a notion of the type of the tabular it's working with. 
So I would leave it as is, but would be interested in your thoughts.

Reply:
Hm, no strong feelings, this interface may also not necessarily dictate how we store tuples, right? That'd depend on the impls, in general I also agree that tables and views are kinda different so treating them as such makes sense